src/DotLiquid.Tests/DropTests.cs

```diff
@@ -397,5 +411,24 @@ public void TestRubyNamingConventionPrintsHelpfulErrorIfMissingPropertyWouldMatc
                 anonymousObject: new { value = new CamelCaseDrop() },
                 namingConvention: new RubyNamingConvention());
         }
+
+        [Test]
+        public void TestTypeResolutionDuplicateNames()
+        {
+            Helper.LockTemplateStaticVars(new RubyNamingConvention(), () =>
+            {
+                var type = typeof(ConflictingChildDrop);
+                var resolver = new TypeResolution(type, mi => true);
+                Assert.IsTrue(resolver.CachedMethods.ContainsKey("get_class_name"));

```

Comment:
Maybe CollectionAssert.Contains on resolver.CachedMethods.Keys. Should give detailed error of the collection not containing the expected key when test fails.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.collectionassert.contains?view=visualstudiosdk-2022

Reply:
I'm not fully understanding your response to the suggestion. I thought I was just suggesting a refactoring of the assert implementation so it validates the same condition but outputs a more detailed message if the condition fails. Did my suggestion change the asserted condition?

Reply:
CollectionAssert.IsSubsetOf would work, but CollectionAssert.Contains returns this:

    Expected: some item equal to < "get_class_name" >
      But was:  < "get_class_name", "to_liquid", "to_string", "get_hash_code", "get_type" >

Reply:
It looks like we have some kind of format massaging missing, as we do see from the contains output that the collection does contain the item as far as we humans can tell.

Reply:
My am I tired. My mistake, it's resolved now.